_posts/2024-06-27-updating-nvidia-jetson-with-memfault.md

```diff
@@ -0,0 +1,312 @@
+$ tar xpf jetson_linux_r36.3.0_aarch64.tbz2
+```
+
+Then, customize the list of packages that are included in the image:
+
+```bash
+$ cd ~/nvidia/Linux_for_Tegra/tools/samplefs
+$ cp nvubuntu-jammy-minimal-aarch64-packages nvubuntu-jammy-interrupt-aarch64-packages
+$ vi nvubuntu-jammy-interrupt-aarch64-packages
+# ... customize the list of packages as needed
+
+# For OTA, we will use curl and jq
+$ echo "curl" >> nvubuntu-jammy-interrupt-aarch64-packages
+$ echo "jq" >> nvubuntu-jammy-interrupt-aarch64-packages
+```
+
+Finally, use the `nv_build_samplefs.sh` script to generate a tarball of the root filesystem with all the packages installed:
+
+```bash
+$ sudo ./nv_build_samplefs.sh --abi aarch64 --distro ubuntu --flavor interrupt --version jammy
+```
+
+> This script only works on Ubuntu 22.04. If you are using a more recent
+> version, [NVIDIA provides a Docker image](https://docs.nvidia.com/jetson/archives/r36.3/DeveloperGuide/SD/RootFileSystem.html#execute-the-script-on-a-non-ubuntu-22-04-host) that you can use to
+> generate the root filesystem.
+
+At this point, you have a `sample_fs.tbz2` file. You need to extract it in the `rootfs` folder of the BSP and run the NVIDIA `apply_binaries.sh` script.
+
+```
+# Decompress in the `rootfs` folder
+$ cd ~/nvidia/Linux_for_Tegra/rootfs
+$ sudo tar xpf ../tools/samplefs/sample_fs.tbz2
+$ cd ..
+
+# Install NVidia binary packages into the rootfs
+$ ./apply_binaries.sh
+```
+
+### Skip `oem-config` and add a default user
+
+For a production image, you will need to add a user account. This will also skip
+the `oem-config` program that is otherwise run automatically by Jetson Linux.
+
+```bash
+$ sudo tools/l4t_create_default_user.sh -u mf -n orin --accept-license
+Creating: Username - mf, Password - OTk5MGE0, Autologin - false
+Creating: Hostname - orin
+```
+
+> A password is automatically generated for you by default. You can also set one
+with the `-p` option. Either way, it would not be a good idea to ship an image
+with a password that someone could possibly find and compromise all your
+devices.
+>
+> For a better alternative, look at our list of [EOSS talks recap](/blog/eoss-recap#engineering-secure-ssh-access-for-engineers) which includes "Engineering Secure SSH Access for Engineers".
+
+### Install Memfault monitoring
+
+Now would also be a great time to add Memfault monitoring to your image (make sure to replace YOUR_PROJECT_KEY with a valid project key).
+
+```bash
+curl -sSf https://raw.githubusercontent.com/patwolfe/memfaultd-experimental/main/quickstart.sh | sudo  MEMFAULT_PROJECT_KEY="<YOUR PROJECT KEY>" chroot rootfs/ sh
+```
+
+> **Make sure to replace `<YOUR PROJECT KEY>` with your actual Memfault project key!**
+>
+> This script will show some errors because it is not designed to run in a
+chroot environment. However, it will install the necessary packages and
+configuration files.
```

Comment:
```suggestion
> This script will show some errors because it is not designed to run in a
chroot environment. However, it will install the necessary packages and
configuration files.
```